sources/transaction.move

```diff
@@ -99,13 +99,47 @@ public fun p2pkh_address(output: &Output): vector<u8> {
 		script[23] == OP_EQUALVERIFY &&
 		script[24] == OP_CHECKSIG
     ) {
-		return slice(script, 3, 23)
+		return true
 	};
-    vector[]
+    false
 }
 
-public fun amount(output: &Output): u256 {
-    output.amount
+public fun is_op_return(output: &Output): bool {
+    let script = output.script_pubkey;
+    return script.length() > 0 && script[0] == OP_RETURN
+}
+
+public fun p2pkh_address(output: &Output): vector<u8> {
+    // TODO: we support P2PKH and P2PWKH now.
+    // We will and more script after.
+    // and the script must return error if we don't support standard script
+
+    let script = output.script_pubkey;
+	return slice(script, 3, 23)
+}
+
+/// Return message after OP_RETURN
+public fun op_return_message(output: &Output): vector<u8> {
+    // If transaction mined to BTC.
```

Comment:
maybe lets move it as a function comment. I think we should explain a little what the function is doing, what is not doing and what are the assumptions.

Reply:
Something like: 
```
- OP_PUSHDATA1: The length is encoded in one byte.
- OP_PUSHDATA2: The length is encoded in two bytes.
- OP_PUSHDATA4: The length is encoded in four bytes.
```

Reply:
Please check!